demisto_sdk/commands/common/hook_validations/base_validator.py

```diff
@@ -154,6 +154,7 @@ def handle_error(
         suggested_fix=None,
         warning=False,
         drop_line=False,
+        ignored_errors=None,
```

Comment:
can we remove `ignored_errors`'s default `None`, to prevent such bugs from happening again?

Reply:
but then every validation that uses it would require to pass in the `ignored_files` even though its already initialized in the constructor of `BaseValidator`. 

This argument has been added in cases where we have `BaseValidator` that was not initialized with `ignored_errors`

Reply:
Do we have any way to check for cases where it's not passed, and not in the parent? (except for adding an `if` everywhere)

Reply:
working on it

Reply:
Well, after carefully viewing it, it seems like we have a few issues to force developers to pass in the `ignored errors`:

1) There are some cases where we use the `BaseValidator` not only to handle validation errors, but to other purposes which does not involve the `ignored_errors`

2) implementing such option causes a lot of unit-tests to fail since they do not pass into the validators constructors the `ignored_errors`.

3) I think because of (1) - we cannot force a developer to pass in the `ignored_errors` into the base-validator, devs should know what they are doing. 

**general note**: i think we should highly consider refactoring the code involves validations because its very hard to read.

Reply:
Thank you, agreed.